add-patch.c

```diff
@@ -1352,6 +1352,15 @@ static int patch_update_file(struct add_p_state *s,
 	struct child_process cp = CHILD_PROCESS_INIT;
 	int colored = !!s->colored.len, quit = 0;
 	enum prompt_mode_type prompt_mode_type;
+	enum {
+		ALLOW_GOTO_PREVIOUS_HUNK = 1 << 0,
+		ALLOW_GOTO_PREVIOUS_UNDECIDED_HUNK = 1 << 1,
+		ALLOW_GOTO_NEXT_HUNK = 1 << 2,
+		ALLOW_GOTO_NEXT_UNDECIDED_HUNK = 1 << 3,
+		ALLOW_SEARCH_AND_GOTO = 1 << 4,
+		ALLOW_SPLIT = 1 << 5,
+		ALLOW_EDIT = 1 << 6
+	} permitted = 0;
```

Comment:
Nice idea.

It might make sense to split this commit up into two commits, though: one that avoids duplicating the permission checking, and then the one that fixes the `e` case (by adding `ALLOW_EDIT`).

Reply:
> It might make sense to split this commit up into two commits, though: one that avoids duplicating the permission checking, and then the one that fixes the `e` case (by adding `ALLOW_EDIT`).

I know what you mean - the bug fix gets a bit lost in all the other changes, but it seems a bit strange to convert everything apart from edit first - I'm not sure ...